packages/sit-onyx/src/components/OnyxDatePicker/OnyxDatePicker.vue

```diff
@@ -0,0 +1,150 @@
+   */
+  "update:modelValue": [value?: string];
+  /**
+   * Emitted when the validity state of the input changes.
+   */
+  validityChange: [validity: ValidityState];
+}>();
+
+const { vCustomValidity, errorMessages } = useCustomValidity({ props, emit });
+const successMessages = computed(() => getFormMessages(props.success));
+const messages = computed(() => getFormMessages(props.message));
+const { densityClass } = useDensity(props);
+const { disabled, showError } = useFormContext(props);
+const skeleton = useSkeletonContext(props);
+const errorClass = useErrorClass(showError);
+
+/**
+ * Gets the normalized date based on the input type that can be passed to the native HTML `<input />`.
+ * Will be checked to be a valid date.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Date_and_time_formats#date_strings
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Date_and_time_formats#local_date_and_time_strings
+ */
+const getNormalizedDate = computed(() => {
+  return (value?: DateValue) => {
+    const date = value != undefined ? new Date(value) : undefined;
+    if (!isValidDate(date)) return;
+
+    const dateString = date.toISOString().split("T")[0];
+    if (props.type === "date") return dateString;
+
+    // for datetime type, the hour must be in the users local timezone so just returning the string returned by `toISOString()` will be invalid
+    // since the timezone offset is missing then
+    const hours = date.getHours().toString().padStart(2, "0");
+    const minutes = date.getMinutes().toString().padStart(2, "0");
+    return `${dateString}T${hours}:${minutes}`;
+  };
+});
+
+/**
+ * Current value (with getter and setter) that can be used as "v-model" for the native input.
+ */
+const value = computed({
+  get: () => getNormalizedDate.value(props.modelValue),
+  set: (value) => {
+    const newDate = new Date(value ?? "");
+    emit("update:modelValue", isValidDate(newDate) ? newDate.toISOString() : undefined);
+  },
+});
+</script>
+
+<template>
+  <div v-if="skeleton" :class="['onyx-datepicker-skeleton', densityClass]">
+    <OnyxSkeleton v-if="!props.hideLabel" class="onyx-datepicker-skeleton__label" />
+    <OnyxSkeleton class="onyx-datepicker-skeleton__input" />
+  </div>
+
+  <div v-else :class="['onyx-datepicker', densityClass, errorClass]">
+    <OnyxFormElement
+      v-bind="props"
+      :error-messages="errorMessages"
+      :success-messages="successMessages"
+      :message="messages"
+    >
+      <template #default="{ id: inputId }">
+        <div class="onyx-datepicker__wrapper">
+          <OnyxLoadingIndicator
+            v-if="props.loading"
+            class="onyx-datepicker__loading"
+            type="circle"
+          />
+          <!-- key is needed to keep current value when switching between date and datetime type -->
+          <input
+            :id="inputId"
+            :key="props.type"
+            v-model="value"
```

Comment:
Is there a particular reason why we don't use [`valueAsDate`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/valueAsDate)? Then we don't need to do the conversion magic ourselves 😅

Reply:
Because valueAsDate does not work for `type="datetime-local"`:
![image](https://github.com/user-attachments/assets/cff5352f-9f65-4c5e-a29e-6bbec59c51fd)

Reply:
So we can not use it for setting the value, but I updated the code to call `valueAsDate` for the `@input` event